src/containers/Widgets/Widgets.js

```diff
@@ -28,7 +28,10 @@ export default class Widgets extends Component {
   static reduxAsyncConnect(params, store) {
     const {dispatch, getState} = store;
     if (!isLoaded(getState())) {
-      return dispatch(loadWidgets());
+      const promise = dispatch(loadWidgets());
+      if (__SERVER__) {
```

Comment:
I thought... may be for this purposes we can simply create new function that will wrap promise and return promise or undefined, depends where we are - on server or on client...